packages/contracts/src/plugins/governance/admin/Admin.sol

```diff
@@ -2,22 +2,21 @@
 
 pragma solidity 0.8.17;
 
-import {SafeCast} from "@openzeppelin/contracts/utils/math/SafeCast.sol";
+import {SafeCastUpgradeable} from "@openzeppelin/contracts-upgradeable/utils/math/SafeCastUpgradeable.sol";
 
-import {Proposal} from "../../../core/plugin/proposal/Proposal.sol";
-import {IMembershipContract} from "../../../core/plugin/membership/IMembershipContract.sol";
+import {ProposalUpgradeable} from "../../../core/plugin/proposal/ProposalUpgradeable.sol";
+import {IMembership} from "../../../core/plugin/membership/IMembership.sol";
 import {PluginCloneable} from "../../../core/plugin/PluginCloneable.sol";
 import {IDAO} from "../../../core/dao/IDAO.sol";
 
 /// @title Admin
 /// @author Aragon Association - 2022-2023
 /// @notice The admin governance plugin giving execution permission on the DAO to a single address.
-contract Admin is IMembershipContract, PluginCloneable, Proposal {
-    using SafeCast for uint256;
+contract Admin is IMembership, PluginCloneable, ProposalUpgradeable {
+    using SafeCastUpgradeable for uint256;
 
     /// @notice The [ERC-165](https://eips.ethereum.org/EIPS/eip-165) interface ID of the contract.
-    bytes4 internal constant ADMIN_INTERFACE_ID =
-        this.initialize.selector ^ this.executeProposal.selector;
```

Comment:
No initialize selector now? We use ProposalUpgradeable while we also use PluginCloneable here

Reply:
> No initialize selector now

After some back and forth, @novaknole and I agree that it should be there.

> We use ProposalUpgradeable while we also use PluginCloneable here

Yes, for reasons described [here by OZ](https://forum.openzeppelin.com/t/does-a-minimal-proxy-contract-that-creates-clones-need-an-upgradeable-smart-contract/29453/3), we must use upgradeable contracts for our cloneable plugins (currently only Admin).